main/EDInitMod.F90

```diff
@@ -278,175 +293,334 @@ subroutine set_site_properties( nsites, sites,bc_in )
+                do el=1,num_elements
+                   call newp%litter(el)%InitConditions(init_leaf_fines=0._r8, &
+                        init_root_fines=0._r8, &
+                        init_ag_cwd=0._r8, &
+                        init_bg_cwd=0._r8, &
+                        init_seed=0._r8,   &
+                        init_seed_germ=0._r8)
+                end do
+
+                sitep => sites(s)
+                if(hlm_use_sp.eq.itrue)then
+                   if(nocomp_pft.ne.0)then !don't initialize cohorts for SP bare ground patch
+                      call init_cohorts(sitep, newp, bc_in(s))
+                   end if
+                else ! normal non SP case always call init cohorts
+                   call init_cohorts(sitep, newp, bc_in(s))
+                end if
+             end if
+          end do !no new patches
+
+          !check if the total area adds to the same as site area
+          tota = 0.0_r8
+          newp => sites(s)%oldest_patch
+          do while (associated(newp))
+             tota=tota+newp%area
+             newp=>newp%younger
+          end do
+
+          if(abs(tota-area).gt.nearzero*area)then
+             if(abs(tota-area).lt.1.0e-10_r8)then ! this is a precision error
+                if(sites(s)%oldest_patch%area.gt.(tota-area+nearzero))then 
+                   ! remove or add extra area 
+                   ! if the oldest patch has enough area, use that
+                   sites(s)%oldest_patch%area = sites(s)%oldest_patch%area - (tota-area)
+                   write(*,*) 'fixing patch precision - oldest',s, tota-area
+                else ! or otherwise take the area from the youngest patch. 
+                   sites(s)%youngest_patch%area = sites(s)%oldest_patch%area - (tota-area)  
+                   write(*,*) 'fixing patch precision -youngest ',s, tota-area
+                endif
+             else !this is a big error not just a precision error. 
+                write(*,*) 'issue with patch area in EDinit',tota-area,tota
+                call endrun(msg=errMsg(sourcefile, __LINE__))
+             endif  ! big error
+          end if ! too much patch area
+
+          ! For carbon balance checks, we need to initialize the 
+          ! total carbon stock
+          do el=1,num_elements
+             call SiteMassStock(sites(s),el,sites(s)%mass_balance(el)%old_stock, &
+                  biomass_stock,litter_stock,seed_stock)
+          end do
+
+          call set_patchno(sites(s))
+
+       enddo !s
+    end if
+
+    ! This sets the rhizosphere shells based on the plant initialization
+    ! The initialization of the plant-relevant hydraulics variables
+    ! were set from a call inside of the init_cohorts()->create_cohort() subroutine
+    if (hlm_use_planthydro.eq.itrue) then 
+       do s = 1, nsites
+          sitep => sites(s)
+          call updateSizeDepRhizHydProps(sitep, bc_in(s))
+       end do
+       deallocate(recall_older_patch)
```

Comment:
`deallocate(recall_older_patch)`
This line should be removed. This line is not existing in "https://github.com/NGEET/fates/blob/master/main/EDInitMod.F90". 

Deallocate will remove the patch that has been created, and cause problems in later steps.  In my case, it complains: 

forrtl: severe (408): fort: (8): Attempt to fetch from allocatable variable FRAGMENTATION_SCALER when it is not allocated

Image              PC                Routine            Line        Source
cesm.exe           000000000408FCA6  Unknown               Unknown  Unknown
cesm.exe           0000000001765F68  fateshistoryinter        2776  FatesHistoryInterfaceMod.F90
cesm.exe           00000000009C3387  clmfatesinterface        1581  clmfates_interfaceMod.F90
cesm.exe           0000000000913EB1  clm_initializemod         746  clm_initializeMod.F90
cesm.exe           0000000000891C4D  lnd_comp_mct_mp_l         234  lnd_comp_mct.F90
cesm.exe           000000000046605F  component_mod_mp_         257  component_mod.F90
cesm.exe           0000000000427AC2  cime_comp_mod_mp_        1350  cime_comp_mod.F90
cesm.exe           000000000045CC59  MAIN__                    122  cime_driver.F90
cesm.exe           0000000000415252  Unknown               Unknown  Unknown
libc-2.17.so       00002AAF381B2505  __libc_start_main     Unknown  Unknown
cesm.exe           0000000000415169  Unknown               Unknown  Unknown